File: 092.cpp
#include <iostream>
#include <vector>
#include <unordered_map>
#include <stack>
using namespace std;

// Definition for singly-linked list.
struct ListNode {
    int val;
    ListNode *next;
    ListNode() : val(0), next(nullptr) {}
    ListNode(int x) : val(x), next(nullptr) {}
    ListNode(int x, ListNode *next) : val(x), next(next) {}
};

class Solution {
public:
    ListNode* reverseBetween(ListNode* head, int left, int right) {
        if (left == right) {
            return head;
        }
        auto dummy = new ListNode();
        dummy->next = head;
        auto p = dummy;
        for (int i = 0; i < left-1; i++) {
            p = p->next;
        }
        auto tail = p->next;
        for (int i = 0; i < right-left; i++) {
            auto tmp = p->next;
            p->next = tail->next;
            tail->next = tail->next->next;
            p->next->next = tmp;
        }
        return dummy->next;
    }
};

int main() {
    std::cout << "result:" << std::endl;
    return 0;
}

File: 092.py
# Definition for singly-linked list.
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

from typing import Optional

class Solution:
    def reverseBetween(self, head: Optional[ListNode], left: int, right: int) -> Optional[ListNode]:
        if left == right:
            return head
        p = dummy = ListNode()
        dummy.next = head # will affect -> p.next = head
        for _ in range(left-1):
            p = p.next
        tail = p.next
        for _ in range(right-left):
            tmp = p.next
            p.next = tail.next
            tail.next = tail.next.next
            p.next.next = tmp
        return dummy.next
